reconcile/utils/terraform/config_provider.py

```diff
@@ -0,0 +1,80 @@
+from dataclasses import dataclass
+from typing import Any, Iterable, Mapping, Optional
+from reconcile.utils.external_resource_spec import ExternalResourceSpecInventory
+from reconcile.utils.external_resources import PROVIDER_AWS, get_external_resource_specs
+from reconcile.utils.ocm import OCMMap
+
+from reconcile.utils.terrascript_aws_client import TerrascriptClient as Terrascript
+
+
+class UnknownProvisionProviderError(Exception):
+    pass
+
+
+@dataclass
+class TerraformConfigProvider:
+    """
+    At a high-level, this class is responsible for generating Terraform configuration in
+    JSON format from app-interface schemas/openshift/external-resource-1.yml objects.
+
+    Usage example (mostly to demonstrate API):
+
+    cp = TerraformConfigProvider("terraform_resources", "qrtf", 20, provisioners, settings)
+    cp.init_spec_inventory(namespaces, provision_provider, provisioner_name)
+    cp.populate_resources(ocm_map=ocm_map)
+    cp.dump(print_to_file, existing_dirs=working_dirs)
+    """
+
+    integration: str
+    integration_prefix: str
+    thread_pool_size: int
+    provisioners: list[dict[str, Any]]
+    settings: Optional[Mapping[str, Any]] = None
+
+    def __post_init__(self):
+        self.ts = Terrascript(
+            self.integration,
+            self.integration_prefix,
+            self.thread_pool_size,
+            self.provisioners,
+            settings=self.settings,
+        )
+
+    def dump(
+        self,
+        print_to_file: Optional[str] = None,
+        existing_dirs: Optional[dict[str, str]] = None,
+    ) -> dict[str, str]:
+        return self.ts.dump(print_to_file, existing_dirs)
+
+    def init_spec_inventory(
+        self, namespaces: Iterable[Mapping[str, Any]], provisioner_name: Optional[str]
```

Comment:
Does this method really need to care about `namespaces`? Or can it just care about specs? It seems like a one-liner for anything calling this to get the specs from the namespaces.

`specs = [get_external_resource_specs(namespace_info) for namespace_info in namespaces]`

It seems weird that namespaces are finding their way into the Terraform code. I understand why from a schema perspective, but I'm wonder if the simple code to extract the specs needs to happen here. The only other thing I see is that `init_populate_specs` also requires namespaces, but it looks like it's just running the same code to filter the specs?